core/src/main/java/io/jenkins/pluginhealth/scoring/scores/AdoptionScoring.java

```diff
@@ -126,73 +106,23 @@ public ScoringComponentResult getScore(Plugin plugin, Map<String, ProbeResult> p
-                    if (days <= Duration.of(365, ChronoUnit.DAYS).toDays()) {
-                        return new ScoringComponentResult(100, getWeight(), List.of("There is less than a year between the last commit and the last release."));
+                    final String defaultReason = "There are %d days between last release and last commit.".formatted(days);
+                    if (days < Duration.of(30 * 6, ChronoUnit.DAYS).toDays()) {
+                        return new ScoringComponentResult(100, getWeight(), List.of(defaultReason, "Less than 6 months gap between last release and last commit."));
                     }
-                    return new ScoringComponentResult(0, getWeight(), List.of("There is no commit within 6 months to 1 year since the last release."));
-                }
-            },
-            new TimeSinceLastCommitScoringComponent() {
-                @Override
-                public String getDescription() {
-                    return "There must be a commit between 1 year and 2 years since the last release.";
-                }
-
-                @Override
-                public ScoringComponentResult getScore(Plugin plugin, Map<String, ProbeResult> probeResults) {
-                    final ProbeResult probeResult = probeResults.get(LastCommitDateProbe.KEY);
-                    if (probeResult == null || ProbeResult.Status.ERROR.equals(probeResult.status())) {
-                        return new ScoringComponentResult(-100, 100, List.of("Cannot determine the last commit date."));
-                    }
-                    final long days = getTimeBetweenLastCommitAndDate(probeResult.message(), plugin.getReleaseTimestamp().withZoneSameInstant(getZone())).toDays();
-                    if (days <= Duration.of(2 * 365, ChronoUnit.DAYS).toDays()) {
-                        return new ScoringComponentResult(100, getWeight(), List.of("There is less than 2 years between the last commit and the last release."));
-                    }
-                    return new ScoringComponentResult(0, getWeight(), List.of("No commit in the last 2 years."));
-                }
-            },
-            new TimeSinceLastCommitScoringComponent() {
-                @Override
-                public String getDescription() {
-                    return "There must be a commit between 2 years and 4 years since the last release.";
-                }
-
-                @Override
-                public ScoringComponentResult getScore(Plugin plugin, Map<String, ProbeResult> probeResults) {
-                    final ProbeResult probeResult = probeResults.get(LastCommitDateProbe.KEY);
-                    if (probeResult == null || ProbeResult.Status.ERROR.equals(probeResult.status())) {
-                        return new ScoringComponentResult(-100, 100, List.of("Cannot determine the last commit date."));
+                    if (days < Duration.of(30 * 12 + 1, ChronoUnit.DAYS).toDays()) {
```

Comment:
The date computations here are strange.  First the conversion is done using a simple formula like `months *30` or `years*365`, then it's passed to `ChronoUnit` conversion from days to days. Would be nice to either do the conversion with `ChronoUnit.MONTHS` or just use the multiplication, but not both.

Reply:
I would love to do that. The problem is that any other `ChronoUnit` than `DAYS` is considered _estimated_ (see https://github.com/openjdk/jdk/blob/744e0893100d402b2b51762d57bcc2e99ab7fdcc/src/java.base/share/classes/java/time/temporal/ChronoUnit.java#L225-L228). Because of that, you get an exception in https://github.com/openjdk/jdk/blob/744e0893100d402b2b51762d57bcc2e99ab7fdcc/src/java.base/share/classes/java/time/Duration.java#L732-L734 (which is called by the [`ZERO.plus(amount, unit)`](https://github.com/openjdk/jdk/blob/744e0893100d402b2b51762d57bcc2e99ab7fdcc/src/java.base/share/classes/java/time/Duration.java#L311-L313), coming from [`Duration.of(amount, unit)`](https://github.com/openjdk/jdk/blob/744e0893100d402b2b51762d57bcc2e99ab7fdcc/src/java.base/share/classes/java/time/Duration.java#L311-L313)).

So, because anything greater than a day is "estimated". That's why all the duration here are in days.